### DISCUSSION TITLE
dark Overlay only for warn, success and error dialogs...

### DISCUSSION THREAD

**Discussion by madcoda9000** [microsoft/fluentui-blazor#716] [Q&A] [answered]
I am wondering that the dark background overlay is available to warn, error and success dialogs only. For custom and confirm dialogs no dark overlay is shown. Is that by design?

Example warn dialog from demo site:

![image](https://github.com/microsoft/fluentui-blazor/assets/25504739/ffa54e49-b128-48cd-b1f5-85463e62badf)

Example confirm dialog from demo site:

![image](https://github.com/microsoft/fluentui-blazor/assets/25504739/157d6daf-dfec-4436-a908-ad0b2ba2914f)

Is there anything i can do to have the background overlay for confirm dialogs too?

Best Regards
Sascha

**Comment by vnbaaij** [comment, marked as answer]
Yes, this is be design. It is because you can't **dismiss** the confirmation by clicking outside of it. You're supposed to give an answer in the confirmation and custom examples.

This behavior is controlled by the `Modal` parameter (in the `DialogParameters`). For a `MessageBox` the `DialogService` evaluates if the `SecondaryAction` is null or empty. If it is `Modal` is true (==dark background). If it is not `Modal` is false (== light background) => you are supposed to take action by clicking one of the buttons. As alwau=ys you can still dismiss the dialog by pressing 'ESC' key.

**Reply by madcoda9000** [reply]
ok, thx for the explanation.

I did it now like described in the "Dialog without Dialogservice" example.

![image](https://github.com/microsoft/fluentui-blazor/assets/25504739/c64055b4-af46-42ce-958b-5fc7e5c15347)
